spring-kafka/src/test/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapterTests.java

```diff
@@ -84,4 +122,16 @@ public void test(Acknowledgment ack) {
 
 	}
 
+	public CompletableFuture<String> future(String data, Acknowledgment ack) {
+
+		CompletableFuture<String> future = new CompletableFuture<>();
+		future.complete("processed" + data);
+		return future;
```

Comment:
There is `completedFuture()` factory to have it as just a single line of code.
Also: is that intentional to not have a white space between `processed` word and data connected to it?